cpp/src/binaryop/binaryop.cpp

```diff
@@ -419,25 +424,50 @@ bool is_same_scale_necessary(binary_operator op)
 std::unique_ptr<column> fixed_point_binary_operation(scalar const& lhs,
                                                      column_view const& rhs,
                                                      binary_operator op,
+                                                     thrust::optional<cudf::data_type> output_type,
                                                      rmm::cuda_stream_view stream,
                                                      rmm::mr::device_memory_resource* mr)
 {
   using namespace numeric;
 
   CUDF_EXPECTS(is_supported_fixed_point_binop(op), "Unsupported fixed_point binary operation");
-  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(),
-               "Both columns must be of the same fixed_point type");
+  CUDF_EXPECTS(lhs.type().id() == rhs.type().id(), "Data type mismatch");
+  CUDF_EXPECTS(op != binary_operator::TRUE_DIV || output_type.has_value(),
+               "TRUE_DIV requires result_type.");
 
-  auto const scale       = compute_scale_for_binop(op, lhs.type().scale(), rhs.type().scale());
-  auto const output_type = is_comparison_binop(op) ? data_type{type_id::BOOL8}  //
-                                                   : data_type{lhs.type().id(), scale};
-  auto out = make_fixed_width_column_for_output(lhs, rhs, op, output_type, stream, mr);
+  auto const scale = op == binary_operator::TRUE_DIV
```

Comment:
This logic doesn't seem to square exactly with the assert above
```
CUDF_EXPECTS(op != binary_operator::TRUE_DIV || output_type.has_value(),
               "TRUE_DIV requires result_type.");
```
The assert (the logic, not the message) potentially implies that if `output_type.has_value() ` you would be using that for the scale even for binops other than TRUE_DIV.

Same applies for the related overloads below.

Reply:
Hmmm, @nvdbaranec I see your point. It was my translation of:
```cpp
if (op == binary_operator::TRUE_DIV)
  CUDF_EXPECTS(output_type.has_value(), "TRUE_DIV requires result_type.");
```
Is there a more clear translation? Or should I use this code :arrow_up:

Reply:
I suppose you could do:
```
CUDF_EXPECTS((op == binary_operator::TRUE_DIV && output_type.has_value()) ||
                         (op != binary_operator::TRUE_DIV),
               "TRUE_DIV requires result_type.");
```

Although at that point, I think putting it inside an if() like you have above is more readable.

Reply:
Agree the if is clearest, even though Conor's translation to operators looks correct.

Reply:
Switched to the `if` version.